### ISSUE TITLE
Angular 17 doesn't build with native-federation

### ISSUE THREAD

**Issue by pamartinez-signify** [angular-architects/module-federation-plugin#377]
While native-federation version 17 does seem to work with ng serve, the build command never finishes.
On a blank Angular 17 project, after running the schematic:
`ng g @angular-architects/native-federation:init --project app-test --port 4201 --type remote `

The first time `ng build` is executed, it shows the following output, and hangs:
```
Application bundle generation complete. [1.645 seconds]
 INFO  Building federation artefacts
 INFO  Preparing shared npm packages
 NOTE  This only needs to be done once, as results are cached
 NOTE  Skip packages you don't want to share in your federation config
```

After cancelling the execution and running a second time, it will stay hung with the following output:
```
Application bundle generation complete. [1.458 seconds]
 INFO  Building federation artefacts
```

This means the project can't successfully build on a CI/CD pipeline

**Comment by manfredsteyer** [comment]
Thanks for pointing this out. 
I did not manage to reproduce this issue
Can you please reproduce this issue with a minimal application?
If not, is there a way to have a look at your source code?

**Comment by manfredsteyer** [comment]
Thanks for pointing this out. This issue is resolved now. 

Please update to the latest version and try again.

**Comment by pamartinez-signify** [comment]
I can confirm that the new version 17.0.2 solves the issue of `npm run build` not returning.
Thank you!

**Comment by pamartinez-signify** [comment]
I see a new DeprecationWarning from node when `npm run build` is executed a second time.

![image](https://github.com/angular-architects/module-federation-plugin/assets/150499920/222064d7-972f-426e-9b71-901dc308083d)

**Comment by pamartinez-signify** [comment]
This was fixed with current version 17.0.3
Thanks!
